### PR TITLE
mon,mgr: add safety check to 'osd destroy'; deprecate 'osd rm'

### PR DESCRIPTION
**Pull request by liewegas**
- osd destroy and osd purge now do the safe-to-destroy safety check.
- if the check fails, --force can be passed to proceed anyway.
- osd rm is deprecated (it has no real purpose at this point)

### CODE REVIEW

**Comment by liewegas** [comment]
@jcsp does the hidden command approach look okay to you?

**Comment by jcsp** [comment]
Yeah, I'm comfortable with "hidden" as a way of flagging internal inter-daemon commands.

It seems like the --yes-i-really-mean-it part could go away on commands that are used this way.

**Comment by liewegas** [comment]
updated to also work with the old --yes-i-really-mean-it

File: src/mgr/MgrCommands.h [hunk 1]
```diff
@@ -107,6 +107,20 @@ COMMAND("osd test-reweight-by-pg " \
 	"dry run of reweight OSDs by PG distribution [overload-percentage-for-consideration, default 120]", \
 	"osd", "r", "cli,rest")
 
+COMMAND("osd destroy "	    \
+        "name=id,type=CephOsdName " \
+        "name=sure,type=CephString,req=False",
```
Comment (tchaikov) [inline src/mgr/MgrCommands.h]: nit, could use `CephChoices`

**Pull request merged**